app/views/ListView.scala.html

```diff
@@ -55,35 +55,45 @@ <h1 class="govuk-heading-l">@renderMessage(viewmodel.heading)</h1>
                 <h1 class="govuk-heading-l">@renderMessage(viewmodel.optViewOnlyDetails.get.heading)</h1>
             }
 
-            @defining(viewmodel.page.paginatedViewModel.fold(viewmodel.page.rows)(paginatedViewModel =>
-                viewmodel.page.rows.paginate(paginatedViewModel.pagination.currentPage, paginatedViewModel.pagination.pageSize))
-            ) { rows =>
-                @hmrcListWithActions(ListWithActionsViewModel(
-                    rows.map(value =>
-                        ListWithActionsItemViewModel(
-                            renderMessage(value.text),
-                            value.change.map(change => ListWithActionsActionViewModel(messages("site.change"), change.url)
-                            .withVisuallyHiddenText(change.hiddenText.toMessage)).toList ++
-                                    value.remove.map(remove => ListWithActionsActionViewModel(messages("site.remove"), remove.url)
-                                    .withVisuallyHiddenText(remove.hiddenText.toMessage)).toList ++
-                                    value.view.map {
-                                        case ListRowLink(url, hiddenText) => ListWithActionsActionViewModel(messages("site.view"), url)
-                                        .withVisuallyHiddenText(hiddenText.toMessage)
-                                        case ListRowNoLink(text) => ListWithActionsActionViewModel.noLink(renderMessage(text))
-                                    }.toList
+            @viewmodel.page.sections.map { section =>
```

Comment:
iterate through each section and renders the rows. also optionally render a heading is it exists